### ISSUE TITLE
IDE/Checkstyle error when editing build configuration

### ISSUE THREAD

**Issue by sishbi** [jshiell/checkstyle-idea#133]
When I edit a build configuration or perform other IDE operations that include a code editor field (such as refactor operation) I see an error reported in the IDE.
We have a couple of custom checks and the error seems to indicate that there is a ClassNotFound exception when loading the custom checker.
Could it be that the custom classes are not present when checkstyle is trying to analyse the editor field?
I see this mostly when I have a Cucumber build configuration, but I am sure I have seen this for other types.

Here is the error and stack trace:
```
2015-02-23 21:35:34,794 [47680946]  ERROR - heckstyle.CheckStyleInspection - The inspection could not be executed. 
org.infernus.idea.checkstyle.exception.CheckStylePluginException: The inspection could not be executed.
	at org.infernus.idea.checkstyle.CheckStylePlugin.processError(CheckStylePlugin.java:120)
	at org.infernus.idea.checkstyle.CheckStyleInspection.checkFile(CheckStyleInspection.java:238)
	at com.intellij.codeInspection.LocalInspectionTool$1.visitFile(LocalInspectionTool.java:144)
	at com.intellij.psi.impl.source.PsiCodeFragmentImpl.accept(PsiCodeFragmentImpl.java:178)
	at com.intellij.codeInspection.InspectionEngine.acceptElements(InspectionEngine.java:74)
	at com.intellij.codeInspection.InspectionEngine.createVisitorAndAcceptElements(InspectionEngine.java:63)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.a(LocalInspectionsPass.java:356)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.access$000(LocalInspectionsPass.java:74)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass$3.process(LocalInspectionsPass.java:325)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass$3.process(LocalInspectionsPass.java:322)
	at com.intellij.concurrency.ApplierCompleter.a(ApplierCompleter.java:119)
	at com.intellij.concurrency.ApplierCompleter.access$000(ApplierCompleter.java:42)
	at com.intellij.concurrency.ApplierCompleter$1.run(ApplierCompleter.java:82)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1154)
	at com.intellij.concurrency.ApplierCompleter$2.run(ApplierCompleter.java:91)
	at com.intellij.openapi.progress.ProgressManager.executeProcessUnderProgress(ProgressManager.java:209)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:212)
	at com.intellij.concurrency.ApplierCompleter.a(ApplierCompleter.java:103)
	at com.intellij.concurrency.ApplierCompleter.compute(ApplierCompleter.java:79)
	at jsr166e.CountedCompleter.exec(CountedCompleter.java:684)
	at jsr166e.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at jsr166e.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:858)
	at jsr166e.ForkJoinPool.scan(ForkJoinPool.java:1687)
	at jsr166e.ForkJoinPool.runWorker(ForkJoinPool.java:1642)
	at jsr166e.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:109)
Caused by: org.infernus.idea.checkstyle.exception.CheckStylePluginException: <html><b>The CheckStyle rules file could not be loaded.</b><br>cannot initialize module TreeWalker - Unable to instantiate oracle.internal.sdp.checkstyle.custom.NoNestedReturnStmts</html>
	at org.infernus.idea.checkstyle.checker.CheckerFactory.blacklistAndShowMessage(CheckerFactory.java:234)
	at org.infernus.idea.checkstyle.checker.CheckerFactory.createChecker(CheckerFactory.java:188)
	at org.infernus.idea.checkstyle.checker.CheckerFactory.getOrCreateCachedChecker(CheckerFactory.java:98)
	at org.infernus.idea.checkstyle.checker.CheckerFactory.getConfig(CheckerFactory.java:147)
	at org.infernus.idea.checkstyle.CheckStyleInspection.getConfig(CheckStyleInspection.java:148)
	at org.infernus.idea.checkstyle.CheckStyleInspection.scanFile(CheckStyleInspection.java:254)
	at org.infernus.idea.checkstyle.CheckStyleInspection.checkFile(CheckStyleInspection.java:225)
	... 23 more
```

**Comment by jshiell** [comment]
I haven't managed to reproduce this yet. What refactoring were you using in particular, just so I can get as close as possible?

I have a suspicion as to what's causing it though - the code path above depends on finding the module for the selected file. If this lookup returns null, the module class loader (which includes the third party JARs isn't creating). I'm guessing that code blocks in dialogues aren't associated with modules, and hence the class loader isn't getting created. It would certainly give the results above, so I'll get to fixing that shortly.

**Comment by jshiell** [comment]
I've fixed that, and placed a [test build on Dropbox](https://dl.dropboxusercontent.com/u/5423904/checkstyle-idea-4.13.2.zip). I'd be most grateful if you could have a try of it and let me know if it solves your problem.

**Comment by sishbi** [comment]
Thanks for the quick fix. I will give the build a try tomorrow.

On Sunday, 1 March 2015, James Shiell <notifications@github.com> wrote:

> I've fixed that, and placed a test build on Dropbox
> <https://dl.dropboxusercontent.com/u/5423904/checkstyle-idea-4.13.2.zip>.
> I'd be most grateful if you could have a try of it and let me know if it
> solves your problem.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jshiell/checkstyle-idea/issues/133#issuecomment-76604492>
> .
>


-- 
Simon Billingsley
simon@sishbi.me.uk
simon@billingsley.me.uk

**Comment by jshiell** [comment]
I'm assuming it hasn't blown up horribly for you, so I've released 4.13.2 with this fix. If I'm wrong, please reopen the bug and I'll have another look :smile:

**Comment by sishbi** [comment]
Sorry for not responding earlier, you caught me at a very busy time in my project.
I have now tried the new version of the plugin and it fixes my original issue.
However this has now exposed something else... I get a checkstyle error because the code snippet (int return type in this case) is not well formed but I guess this is a new issue.
See attached image.
![refactor_dialog_cs_error](https://cloud.githubusercontent.com/assets/3800191/6922791/9cb800cc-d7c5-11e4-9b3d-ae98ff261347.png)

**Comment by jshiell** [comment]
Apologies - I forgot to reopen this issue, and completely forgot about this dialogue issue.

The good news is that despite my foolishness it's now fixed in the latest release: as part of #172 I included checks to see that the scanned file was in the source directories, which avoids problems like this.
